src/hexen/r_things.c

```diff
@@ -660,6 +667,18 @@ void R_ProjectSprite(mobj_t * thing)
     }
 
     vis->brightmap = R_BrightmapForSprite(thing->state - states);
+#ifdef CRISPY_TRUECOLOR
+    // [crispy] not using additive blending (I_BlendAdd) here for full
+    // bright states to preserve look & feel of original Hexen's translucency
+    if (thing->flags & MF_SHADOW)
+    {
+        vis->blendfunc = I_BlendOverTinttab;
+    }
+    if (thing->flags & MF_ALTSHADOW)
```

Comment:
is this an `else if` or is it possible that a thing carries both flags? Would `MF_ALTSHADOW` take precedence in that case?

Reply:
Nice catch, thank you! Apparently, here must be `else if` condition type, if compared to [original code](https://github.com/fabiangreffrath/crispy-doom/blob/master/src/hexen/r_things.c#L401-L418) handling. Original should represent logics:
* If `MF_SHADOW` _and_ `MF_ALTSHADOW` flags are set, then use `tlcolfunc` (not `alttlcolfunc`).

Honestly, no idea is it possible to achieve such effect in the game, but I can set both flags to some mobj and see what will happen, in both paletted and truecolor renders, that's should the only ideal test.

Reply:
Yes, just as expected. In paletted render, if both flags are set (matters no in which order), then lower opacity `tlcolfunc` will be used. 

- [PAL - 1 MF_SHADOW](https://github.com/fabiangreffrath/crispy-doom/assets/21193394/575ad294-da8b-4f82-af33-8ce9bdfbe2eb)
- [PAL - 2 - MF_ALTSHADOW](https://github.com/fabiangreffrath/crispy-doom/assets/21193394/a55865ab-b6c8-4d72-b524-6b774e45a18e)
- [PAL - 3 - MF_SHADOW | MF_ALTSHADOW](https://github.com/fabiangreffrath/crispy-doom/assets/21193394/f21dea10-fa89-421b-9dda-0befc363019c)
- [PAL - 4 - MF_ALTSHADOW | MF_SHADOW](https://github.com/fabiangreffrath/crispy-doom/assets/21193394/4d95a8d7-fd13-4418-a376-9242ca434ca6)

In current TC implementation it's different, second `if (thing->flags & MF_ALTSHADOW)` overrides first one, which is leading to more opaque sprites in case of both flags are set:

- [TC - 1 - MF_SHADOW](https://github.com/fabiangreffrath/crispy-doom/assets/21193394/c83f882f-e6d3-4c4c-9442-58d18456b9a8)
- [TC - 2 - MF_ALTSHADOW](https://github.com/fabiangreffrath/crispy-doom/assets/21193394/181bfff0-1d13-4b08-aa93-4b4f7578b31c)
- [TC - 3 - MF_SHADOW | MF_ALTSHADOW](https://github.com/fabiangreffrath/crispy-doom/assets/21193394/53aeed64-8d44-4a61-bde5-5b02b55e2e95)
- [TC - 4 - MF_ALTSHADOW | MF_SHADOW](https://github.com/fabiangreffrath/crispy-doom/assets/21193394/aedb33e4-7313-4a03-af73-4c64bc1f20ba)

So yeah, of course second condition should be guarded as `else if`, that's will do correct behavior.